cassandra/metadata.py

```diff
@@ -2833,3 +2833,44 @@ def group_keys_by_replica(session, keyspace, table, keys):
             keys_per_host[NO_VALID_REPLICA].append(key)
 
     return dict(keys_per_host)
+
+
+def strip_frozen(cass_type):
+    """
+    Takes a string representation of a cassandra type and removes frozen
+    E.g., 'frozen<tuple<int>>' becomes 'tuple<int>'
+    """
+    scanner = re.Scanner((
+        (r'[a-zA-Z0-9_]+', lambda s, t: t),
+        (r'[<>, ]', lambda s, t: t),
+        (r'".*?"', lambda s, t: t),
+    ))
+
+    scanned_tokens = scanner.scan(cass_type)[0]
+
+    def _strip(tokens):
+        """ Recursively remove each 'frozen' instance (and its brackets) from the list of tokens """
+        if 'frozen' not in tokens:
+            return tokens
+        updated_tokens = []
+        depth = 0
+        in_frozen = False
+        for token in tokens:
+            skip_current_token = False
+            if token == 'frozen' and not in_frozen:
+                in_frozen = True
+                skip_current_token = True
+            elif token == '<' and in_frozen:
+                depth += 1
+                if depth == 1:
+                    skip_current_token = True
+            elif token == '>' and in_frozen:
+                depth -= 1
+                if depth == 0:
+                    in_frozen = False
+                    skip_current_token = True
+            if not skip_current_token:
+                updated_tokens.append(token)
+        return _strip(updated_tokens)
+
+    return ''.join(_strip(scanned_tokens))
```

Comment:
What about using regex? Could something like this work for this case?
```
FROZEN_REMOVAL_REGEX = re.compile(r'frozen<(.+)>')
def strip_frozen(cass_type):
    while 'frozen' in cass_type: 
        cass_type = re.sub(FROZEN_REMOVAL_REGEX, r'\1', cass_type)
    return cass_type
```

Reply:
Unfortunately no, regex can't deal with arbitrary nesting. E.g.,
`frozen<tuple<int, frozen<tuple<int, text>>, int>>`
A non-greedy regex would remove the first end bracket, and a greedy one would remove the last one. Neither are correct.

Reply:
The function is a bit hard to read... not really because how it is done but mostly because it's something messy to do.  Maybe it we be possible to do something in a generic way and split it a bit more. e.g:

1. A function that only takes care to return a list of the types, grouped by container/collection
2. A recursive function that only takes care to return the inner of a cass type. tuple<THIS>, frozen<THIS>.

Just brainstorming.. we can discuss this more. I think it could be useful to have that for the future... since cql_types_from_string is not really smart.

Also could we move this stuff in cqltypes.py?